purchases/src/main/kotlin/com/revenuecat/purchases/DangerousSettings.kt

```diff
@@ -11,7 +11,17 @@ data class DangerousSettings internal constructor(
      */
     val autoSyncPurchases: Boolean = true,
 
+    /**
+     * Do not consume IAP (In app products) after a successful purchase. This can be useful when using IAP as lifetime
+     * products and you don't want them to disappear from the Google query method. DO NOT use this if you allow
+     * purchasing the same IAP multiple times, like for currency/items. This also means that the purchase might be
+     * synced with multiple users in different devices, unless they are logged in with the same user ID.
+     * This does not affect Amazon. This is disabled by default.
+     */
+    val doNotConsumeIAP: Boolean = false,
```

Comment:
Need to add this to API tests. I haven't found DangerousSettings in our current API tests, so I will add it for those.